### PR TITLE
Cluster tab improvements

### PR DESCRIPTION
**Pull request by Gruek**
This pull request contains:
- SLURM batch launchers
- Improved status output for clusters
- Additional inputs for batch clusters

http://nbviewer.ipython.org/gist/Gruek/2fce54a22f3a5a5ae522

### CODE REVIEW

**Comment by minrk** [comment]
Can we separate the new launcher and the UI to separate PRs? I think the two are largely unrelated, and there is a great deal more discussion to be had on the UI than adding a launcher.

File: CSIRO BSD MIT License v2.0.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
+CSIRO Open Source Software License Agreement (variation of the BSD / MIT License)
```
Comment (minrk) [inline CSIRO BSD MIT License v2.0.txt]: Would you be willing to license your contributions under the BSD license that IPython uses? We do not want to have different licenses applying to different parts of IPython.

This file also appears to be full of unicode errors.

**Comment by Gruek** [comment]
I can separate the SLURM launchers to the UI changes, but those UI changes modify the batch launchers a lot, so if I separate out the launchers they won't be compatible with the other changes which would then have to be reapplied if the UI changes get accepted. Slurm is the one I'm working with currently, which is why I included it in the same request and it would be a pain for me to test on it and then strip it out each time in git.

As for the license issue, I will have to check and report back.

**Comment by minrk** [comment]
If you want to keep the changes in IPython.parallel together, that would be fine, too, I just think everything here is a bit much for one discussion. It's easier to review a few things at a time, and this does a lot in a variety of places:

1. add SLURM launcher (AOK, happy to have it)
2. changes to parameters and arguments in batch launchers (some review to go over):
    - new `info` command
    - changes to batch launcher parameters
3. add cluster input UI (separate discussion)

If you would prefer to leave it all as one PR, that's fine, too, it will just take longer for us to get through the review, since there's a lot to go over.

We are about to release IPython 3.0, but we will get back to reviewing more PRs once we ship 3.0.

ping @ellisonbg for cluster input UI review

**Comment by Gruek** [comment]
I would prefer to keep these as one PR for now, if there are issues with implementing some of these chunks, then I will have a look at separating them.

**Comment by ellisonbg** [comment]
I am also +1 on separating the different parts of this.

**Comment by minrk** [comment]
@Gruek Fair enough. We'll get back to this after you've checked on the license and we get out from under the 3.0 release.

**Comment by Carreau** [comment]
FYI need rebase.

**Comment by minrk** [comment]
Since the split has begun, this will need to be issued as separate PRs against parallel and the notebook server.

**Comment by ellisonbg** [comment]
Any plans to continue to this soon? If not, do you mind if we close this until work restarts against our re-organized repos. We are making major changes to our repos right now and existing PRs are going to bit-rot quickly (already).

**Comment by Gruek** [comment]
I'm not sure when I can get around to doing this, so feel free to close this PR.

**Comment by ellisonbg** [comment]
Closing this due to inactivity. Please feel free to reopen as work restarts. I have created an issue to track progress: #8189.

**Pull request closed without merging**